src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanicsEFEMKernel.hpp

```diff
@@ -582,10 +582,68 @@ using SinglePhaseKernelFactory = finiteElement::KernelFactory< SinglePhase,
                                                                real64 const (&)[3],
                                                                arrayView1d< string const > const >;
 
+/**
+ * @brief A struct to perform volume, aperture and fracture traction updates
+ */
+struct StateUpdateKernel
+{
+
+  /**
+   * @brief Launch the kernel function doing volume, aperture and fracture traction updates
+   * @tparam POLICY the type of policy used in the kernel launch
+   * @tparam CONTACT_WRAPPER the type of contact wrapper doing the fracture traction updates
+   * @param[in] size the size of the subregion
+   * @param[in] contactWrapper the wrapper implementing the contact relationship
+   * @param[in] dispJump the displacement jump
+   * @param[in] pressure the pressure
+   * @param[in] deltaPressure the accumulated pressure updates
+   * @param[in] area the area
+   * @param[in] volume the volume
+   * @param[out] deltaVolume the change in volume
+   * @param[out] aperture the aperture
+   * @param[out] effectiveAperture the effecture aperture
+   * @param[out] fractureTraction the fracture traction
+   * @param[out] dFractureTraction_dPressure the derivative of the fracture traction wrt pressure
+   */
+  template< typename POLICY, typename CONTACT_WRAPPER >
+  static void
+  launch( localIndex const size,
+          CONTACT_WRAPPER const & contactWrapper,
+          arrayView2d< real64 const > const & dispJump,
+          arrayView1d< real64 const > const & pressure,
+          arrayView1d< real64 const > const & deltaPressure,
+          arrayView1d< real64 const > const & area,
+          arrayView1d< real64 const > const & volume,
+          arrayView1d< real64 > const & deltaVolume,
+          arrayView1d< real64 > const & aperture,
+          arrayView1d< real64 > const & effectiveAperture,
+          arrayView2d< real64 > const & fractureTraction,
+          arrayView1d< real64 > const & dFractureTraction_dPressure )
+  {
+    forAll< POLICY >( size, [=] GEOSX_HOST_DEVICE ( localIndex const k )
+    {
+      // update aperture to be equal to the normal displacement jump
+      aperture[k] = dispJump[k][0]; // the first component of the jump is the normal one.
```

Comment:
this aperture is probably useless now. As a matter of fact, the `computeEffectiveAperture` should use the `displacementJump` vector and then some models will use the normal component while others will need to use the tangential ones to then come up with and `hydraulicAperture` (which is probably a better name for it). we don't need to change this in this PR though.  Even for the HydrofractureSolver and the Lagrangian one I would first reconstruct the dispJump and then use that in the constitutive models. It well help also with permeability models that need more than just the normal component (see https://github.com/GEOSX/GEOSX/pull/1564)

Reply:
I agree, we should unify all this, happy to help with this work in the next PR.